File: 6th Semester/Programming Paradigms/NP2 Exercises/ex9.c
// Escreva um programa para aprovar o empréstimo bancário para compra de uma casa. O programa
// deve perguntar o valor da casa a comprar, o salário e a quantidade de anos a pagar. O valor da prestação 
// mensal não pode ser superior a 30% do salário. Calcule o valor da prestação como sendo o valor da casa a
// comprar dividido pelo numero de meses a pagar

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <math.h>
#define TAMANHO_OUTPUT 100

int lerInteiro(char mensagem[]) {
    char output[TAMANHO_OUTPUT];
    int inteiro;
    while(1) {
        printf("%s", mensagem);
        fgets(output, TAMANHO_OUTPUT, stdin);
        if (!((inteiro = atoi(output)) == 0 && output[0] != '0'))
            return inteiro;
        printf("Por favor insira um número inteiro válido!\n");
    }
}

float lerFloat(char mensagem[]) {
    char output[TAMANHO_OUTPUT];
    float numFloat;
    while(1) {
        printf("%s", mensagem);
        fgets(output, TAMANHO_OUTPUT, stdin);
        if (!((numFloat = atof(output)) == 0 && output[0] != '0'))
            return numFloat;
        printf("Por favor insira um número real válido!\n");
    }
}

int main() {
    // Compilação do programa no linux: "gcc ex9.c -lm"
    printf("Programa aprova seguro.");
    float valorCasa = lerFloat("Insira o VALOR DA CASA a se comprar: ");
    float salario = lerFloat("Agora, por favor insira o valor do SALÁRIO: ");
    int anos = lerInteiro("Por fim, insira a quantidade de ANOS que se deseja quitar a dívida: ");
    float prestacao = valorCasa / (anos*12);
    if (prestacao > salario*0.3) {
        printf("Emprestimo recusado! O valor de R$%.2f da prestação é maior que R$%.2f correspondente a 30\% do salário.\n", prestacao, salario);
        printf("Por este valor, você precisaria de pelo menos %.0f anos para o empréstimo ser aprovado.\n", ceilf((valorCasa / (salario*3.6))));
        printf("Ou ganhar R$%.2f de salário.\n", prestacao/0.3);
    } else {
        printf("Emprestimo aprovado! Por causa do valor de R$%.2f da prestação ser menor que R$%.2f correspondente a 30\% do salário, você tem direito ao empréstimo.", prestacao, salario*0.3);
    }
    return 0;
}

File: 5th Semester/Programming Paradigms/NP2 Exercises/ex9.py
# Escreva um programa para aprovar o empréstimo bancário para compra de uma casa. O programa
# deve perguntar o valor da casa a comprar, o salário e a quantidade de anos a pagar. O valor da prestação 
# mensal não pode ser superior a 30% do salário. Calcule o valor da prestação como sendo o valor da casa a
# comprar dividido pelo numero de meses a pagar
from math import ceil

def ler_float(mensagem: str) -> float:
    while True:
        output = str(input(mensagem)).strip()
        try:
            return float(output.replace(',', '.'))
        except ValueError:
            print('Por favor insira um número real válido.')

def ler_inteiro(mensagem: str) -> int:
    while True:
        output = str(input(mensagem)).strip()
        if (output.replace('-','').isnumeric()):
            return int(output)
        print('Por favor insira um número inteiro válido.')


def main():
    print("Programa aprova seguro.")
    valor_casa: float = ler_float("Insira o VALOR DA CASA a se comprar: ")
    salario: float = ler_float("Agora, por favor insira o valor do SALÁRIO: ")
    anos: int = ler_inteiro("Por fim, insira a quantidade de ANOS que se deseja quitar a dívida: ")
    prestacao_mensal: float = valor_casa/(anos*12)
    if prestacao_mensal > salario*0.3:
        print(f"Emprestimo recusado! O valor de R${prestacao_mensal:.2f} da prestação é maior que R${salario*0.3:.2f} correspondente a 30% do salário.")
        print(f"Por este valor, você precisaria de pelo menos {ceil(valor_casa / (salario*3.6))} anos para o empréstimo ser aprovado.")
        print(f"Ou ganhar R${prestacao_mensal/0.3:.2f} de salário.")
    else:
        print(f"Emprestimo aprovado! Por causa do valor de R${prestacao_mensal:.2f} da prestação ser menor que R${salario*0.3:.2f} correspondente a 30% do salário, você tem direito ao empréstimo.")
    

if __name__ == "__main__":
    main()